packages/boot/tools/ibc/mocks.js

```diff
@@ -7,22 +52,85 @@
  */
 export const icaMocks = {
   startChannelOpenInit: {
-    // ICA Channel Creation
-    channelOpenAck: obj => ({
-      type: 'IBC_EVENT',
-      blockHeight: 99,
-      blockTime: 1711571357,
-      event: 'channelOpenAck',
-      portID: obj.packet.source_port,
-      channelID: 'channel-0',
-      counterparty: {
-        port_id: obj.packet.destination_port,
-        channel_id: 'channel-1',
-      },
-      counterpartyVersion:
-        '{"version":"ics27-1","controllerConnectionId":"connection-0","hostConnectionId":"connection-0","address":"osmo1234","encoding":"proto3","txType":"sdk_multi_msg"}',
-      connectionHops: obj.hops,
-    }),
+    /**
+     * ICA Channel Creation
+     * @param {ChannelOpenInitPacket} obj
+     */
+    channelOpenAck: obj => {
+      const icaControllerNonce = obj?.packet?.source_port?.split('-')?.[1];
+      return {
+        type: 'IBC_EVENT',
+        blockHeight: 99,
+        blockTime: 1711571357,
+        event: 'channelOpenAck',
+        portID: obj.packet.source_port,
+        channelID: `channel-${icaControllerNonce}`,
+        counterparty: {
+          port_id: obj.packet.destination_port,
+          channel_id: `channel-${icaControllerNonce}`,
+        },
+        counterpartyVersion: addParamsToVersion(obj.version, {
+          address: 'cosmos1test',
+        }),
+        connectionHops: obj.hops,
+      };
+    },
     // XXX channelOpenAckFailure
   },
+  sendPacket: {
+    /**
+     * ICA Send Packet (Transaction) - MsgDelegate success
+     * @param {SendPacket} obj
+     * @param {number} sequence transaction sequence number
+     */
+    msgDelegateResponse: (obj, sequence = 1) => {
+      return {
+        // {"result":"+/cosmos.staking.v1beta1.MsgDelegateResponse"}
```

Comment:
what's this comment meant to convey?

Reply:
These code comments are the decoded acknowledgement packets. I refactored `mocks.js` and `support.ts` per your feedback, lmk if this reads better:

https://github.com/Agoric/agoric-sdk/pull/9195/files#diff-6c46d78f5e5132e1c8596f11536961851d788b0ea13d1d60b47f87faf3a6e436R34-R58

Reply:
yep!